code/game/machinery/doors/firedoor.dm

```diff
@@ -73,10 +73,13 @@
 	acid = 70
 
 /obj/machinery/door/firedoor/Initialize(mapload)
+	id_tag = assign_random_name()
 	. = ..()
 	soundloop = new(src, FALSE)
 	CalculateAffectingAreas()
 	my_area = get_area(src)
+	if(name == initial(name))
+		name = "[get_area_name(my_area)] [initial(name)] [id_tag]"
```

Comment:
should you move this to `update_name()` like we have for fire alarms?

Reply:
Yeah that would improve consistency.